presenters/thread.rb

```diff
@@ -23,7 +23,7 @@ def initialize(thread, user, is_read, unread_count, is_endorsed)
     @is_endorsed = is_endorsed
   end
 
-  def to_hash with_responses=false, resp_skip=0, resp_limit=nil
+  def to_hash with_responses=false, resp_skip=0, resp_limit=nil, recursive=false
```

Comment:
We should change the default to `recursive = True`.  

I checked in edx/edx-platform:lms/djangoapps/discussion_api/api.py, and it looks like this view is called from `_get_thread_and_context`.   There is one call to this function that specifies `recursive = True`, one call that specifies `recursive = False`, and several calls that do not specify anything for the `recursive` option.  This makes me think there will be backwards compatibility issues here.  

While I agree that it would be nice to not use the recursive behavior unless the calling code explicitly opts in, I think correctness trumps cleanliness.